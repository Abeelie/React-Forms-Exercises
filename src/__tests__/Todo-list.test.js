import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDoList from "../Components/todo-list/Todo-List";


it("renders without crashing", function () {
    render(<ToDoList />);
});
  
it("matches snapshot", function () {
    const { asFragment } = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
});



const addTodo = (todoList, name = "random") => {
    const todoLabel = todoList.getByLabelText("Todo:");
  
    fireEvent.change(todoLabel, { target: { value: name } });
  
    const button = todoList.getByText("Add todo");
    fireEvent.click(button);
}


it("adds todo to list", function () {
    const todoList = render(<ToDoList />);
    addTodo(todoList);
  
    const removeBtn = todoList.getByText("x");
    expect(removeBtn).toBeInTheDocument();
    expect(todoList.queryByText("random")).toBeInTheDocument();
});


it("removes todo from list", function () {
    const todoList = render(<ToDoList />);
    addTodo(todoList);
  
    const removeBtn = todoList.getByText("x");
    fireEvent.click(removeBtn);
    expect(todoList.queryByText("random")).not.toBeInTheDocument();
    expect(removeBtn).not.toBeInTheDocument();
  });