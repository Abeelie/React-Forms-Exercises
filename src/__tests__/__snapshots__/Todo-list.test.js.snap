// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      Todo App
    </h1>
    <form
      class="todo-form"
    >
      <label
        for="name"
      >
        Todo:
      </label>
      <input
        id="name"
        name="name"
        value=""
      />
      <button
        id="btn"
      >
        Add todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
